client/components/checklist/index.js

```diff
@@ -16,37 +16,53 @@ import TaskPlaceholder from 'components/checklist/task-placeholder';
 export default class Checklist extends PureComponent {
 	static propTypes = {
 		isPlaceholder: PropTypes.bool,
+		updateCompletion: PropTypes.func,
 	};
 
-	state = { hideCompleted: false };
-
-	toggleCompleted = () =>
-		this.setState( ( { hideCompleted } ) => ( { hideCompleted: ! hideCompleted } ) );
+	componentDidMount() {
+		this.notifyCompletion();
+	}
 
-	renderPlaceholder() {
-		return (
-			<div className={ classNames( 'checklist', 'is-expanded', 'is-placeholder' ) }>
-				<ChecklistHeader total={ 0 } completed={ 0 } />
-				<div className="checklist__tasks">
-					{ times( Children.count( this.props.children ), index => (
-						<TaskPlaceholder key={ index } />
-					) ) }
-				</div>
-			</div>
-		);
+	componentDidUpdate() {
+		this.notifyCompletion();
 	}
 
-	render() {
-		if ( this.props.isPlaceholder ) {
-			return this.renderPlaceholder();
+	notifyCompletion() {
+		if ( 'function' === typeof this.props.updateCompletion ) {
+			const [ complete, total ] = this.calculateCompletion();
+			this.props.updateCompletion( { complete: complete >= total } );
 		}
+	}
 
+	calculateCompletion() {
 		const { children } = this.props;
-
-		const count = Children.map( children, child => child.props.completed ).reduce(
-			( acc, completed ) => ( true === completed ? acc + 1 : acc ),
+		const completedCount = Children.toArray( children ).reduce(
+			( count, task ) => ( true === task.props.completed ? count + 1 : count ),
```

Comment:
Would it make sense to use [`countBy()`](https://lodash.com/docs/4.17.10#countBy) instead of `reduce()` here?

Reply:
It might, but I'm not familiar with `countBy`. If you want to rewrite to a simpler `countBy`, please feel free 👍